### ISSUE TITLE
Feature Request - Dealing with 'sleepy' blinds

### ISSUE THREAD

**Issue by dzerovibe** [mtgeekman/Home_Assistant_NeoSmartBlinds#52]
Thanks for this great integration. I got it up and running as soon as I got my new Luxaflex blinds with the Neo controller. I quickly realised that if the controller/blind was idle for a while, then the next command wouldn't work, and I would have to send the command twice in order to make it work. I got in touch with a developer at Neo, and he said that my particular blind motors have a sort of sleep mode. When they haven't received a command in a while, they go to sleep, and the next received command seems to wake them up, but they don't actually respond to that command.

Can I request an optional feature for 'sleepy' blinds, and if this option is enabled, if a command hasn't been sent after X minutes (configurable) then the new command should be sent twice?

**Comment by JBDenton** [comment]
It's been a little while since I looked at the underlying code but from memory with some work it could be possible to do. Could either have a list of known motor codes that are known to be "lazy" but this could prove to be difficult to maintain or I feel a better option would be to have a flag (possibly optional) for each blind in config for it to be lazy.
 The checking if it's been a period of time since last call to the blind, that could be tricky, would each motor code have the same timeout period or would this likely be something that would be configured independently for each blind?
Did the developer at NeoSmart give any indication of how many models and which kinds are affected?
It's good to know that they are still keen to help HA folk, I was in touch with them ages ago for some work on this integration that I hope to get back to at some point.

**Comment by dzerovibe** [comment]
> It's good to know that they are still keen to help HA folk, I was in touch with them ages ago for some work on this integration that I hope to get back to at some point.

Yes, I was pleasantly surprised to get such responsive and helpful replies when I contacted them. It was somebody called Antonio who answered my emails. I'd say he'd be happy to help you if you got in touch again.

> Did the developer at NeoSmart give any indication of how many models and which kinds are affected?

No, he just asked what my motor code was, and told me based on that.

> a better option would be to have a flag (possibly optional) for each blind in config for it to be lazy.

That's basically what I was thinking; a new option to flag a lazy blind, and then perhaps another option to specify a timeout period.

> would each motor code have the same timeout period or would this likely be something that would be configured independently for each blind?

I imagine there could be a lot of variance between motors, so I think the easiest option would be just to let it be configurable. Perhaps a default value could be coded in. I'll do some further experimenting with mine to try and get an idea of how long it takes for them to go to sleep.

Thanks for for your response and willingness to consider this for me!
